File: GUST/gust-be/src/main/java/gust/model/SugarLog.java
package gust.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "sugar_logs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SugarLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /** grams of sugar consumed */
    private int sugarGrams;

    /** date when logged */
    private LocalDate date;

    /** hour of the day (0–23) */
    private int hour;

    /** minute of the hour (0–59) */
    private int minute;

    /** product or food name */
    private String productName;

    /** type/category of sugar intake */
    private String sugarType;

    /** any contextual notes */
    private String contextNote;

    /** emotional state during logging */
    @Enumerated(EnumType.STRING)
    private Emotion emotion;

    /** location where intake occurred */
    private String location;

    /** whether it was a craving */
    private boolean wasCraving;

    /** the user who logged this entry */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

File: GUST/gust-fe/lib/models/sugar_log.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'emotion.dart';

part 'sugar_log.freezed.dart';
part 'sugar_log.g.dart';

@freezed
class SugarLog with _$SugarLog {
  const factory SugarLog({
    required int id,
    required int sugarGrams,
    required DateTime date,
    required int hour,
    required int minute,
    required String productName,
    required String sugarType,
    String? contextNote,
    required Emotion emotion,
    String? location,
    @Default(false) bool wasCraving,
    int? userId,
  }) = _SugarLog;

  factory SugarLog.fromJson(Map<String, dynamic> json) => _$SugarLogFromJson(json);
}

@freezed
class SugarLogRequest with _$SugarLogRequest {
  const factory SugarLogRequest({
    required int sugarGrams,
    required String date, // Format: YYYY-MM-DD
    required int hour,
    required int minute,
    required String productName,
    required String sugarType,
    String? contextNote,
    required String emotion, // String for backend
    String? location,
    @Default(false) bool wasCraving,
  }) = _SugarLogRequest;

  factory SugarLogRequest.fromJson(Map<String, dynamic> json) => _$SugarLogRequestFromJson(json);
}

@freezed
class SugarLogResponse with _$SugarLogResponse {
  const factory SugarLogResponse({
    required int id,
    required int sugarGrams,
    required String date,
    required int hour,
    required int minute,
    required String productName,
    required String sugarType,
    String? contextNote,
    required String emotion,
    String? location,
    @Default(false) bool wasCraving,
  }) = _SugarLogResponse;

  factory SugarLogResponse.fromJson(Map<String, dynamic> json) => _$SugarLogResponseFromJson(json);
}
